File: Web Alex Roel/s18-Python POO/herencia.py

class Persona:
    def __init__(self, nombre, edad): #Constructor
        self.nombre = nombre
        self.edad = edad 
        
    def detalle_persona(self):
        print(f'Nombre: {self.nombre} \nEdad: {self.edad}')

    def __str__(self): #Imprime
        return f'Nombre: {self.nombre} \nEdad: {self.edad}'

class Cliente(Persona): #Herencia
    pass

File: Web Alex Roel/s10-JS-POO/herencia.js
class Persona{
    
    constructor(nombre , edad){
        this.nombre = nombre;
        this.edad = edad;
    }
    
    detallePersona(){
        console.log(`Nombre: ${this.nombre}\nEdad: ${this.edad}`);
    }

}

class Empleado extends Persona{
    
    sueldo;

    constructor(nombre, edad, sueldo){
        super(nombre,edad);//Hereda el constructor de la clase padre(Persona)
        this.sueldo = sueldo;
    }
    detallePersona(){
        super.detallePersona();
        console.log(`Sueldo: ${this.sueldo}`);
    }

}

const em1 = new Empleado('Pablo',21,500);
console.log(em1);

em1.detallePersona();